src/seedu/addressbook/commands/Command.java

```diff
@@ -55,6 +55,14 @@ public void setData(AddressBook addressBook, List<? extends ReadOnlyPerson> rele
     }
 
     /**
+     * This method will return true for command types that mutate the data. e.g. AddCommand.
+     * @return result on whether data is mutated.
+     */
+    public boolean isMutating() {
+        return false;
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.